src/test/resources/features/Device.feature

```diff
@@ -0,0 +1,75 @@
+Feature: Testing Device API
+
+  Scenario Outline: Create Device
+    Given I want to create a device
+    When I create a device
+    Then I should get a response with status <code>
+    And I should get a response with body
+
+    Examples:
+      | code |
+      | 200  |
+
+  Scenario Outline: Create Same Device
+    Given I want to create a device
+    When I create a device
+    Then I should get a response with status <code>
+
+    Examples:
+      | code |
+      | 409  |
+
+  Scenario Outline: Get Devices
+    Given I want to get all devices
+    When I get all devices
+    Then I should get a response with status <code>
+
+    Examples:
+      | code |
+      | 200  |
+
+  Scenario Outline: Get a Device
+    Given I want to get the device with serialNumber <serialNumber>
+    When I get a device
+    Then I should get a response with status <code>
+    And I should get a response with body
+
+    Examples:
+      | code | serialNumber |
+      | 200  | "1"          |
+
+
+  Scenario Outline: Get a Non-existing Device
+    Given I want to get the device with serialNumber <serialNumber>
+    When I get a device
+    Then I should get a response with status <code>
+
+    Examples:
+      | code | serialNumber |
+      | 404  | "2"          |
+      | 404  | "abc"        |
+
```

Comment:
There are two lines of examples here, will this run as two "test cases"?

Reply:
Yes!! and thats the beauty of cucumber, although my examples don't show this. But basically, you can have a test that runs a lot of examples to test different behaviors:
an example can be found here: https://www.baeldung.com/cucumber-data-tables